docs/sci-integration.md

```diff
@@ -0,0 +1,117 @@
+
+### Namespace Structure
+
+The SCI context includes several namespaces:
+
+1. **`clojure.core`**: Core Clojure functions needed for query expressions
+2. **`fluree.db.query.exec.eval`**: Query evaluation functions like `iri`, `datatype`, `str`, etc.
+3. **`fluree.db.query.exec.where`**: Functions for handling typed values and query matching
+4. **`fluree.json-ld`**: JSON-LD functions, particularly `expand-iri`
+5. **`fluree.db.constants`**: Constants like datatype IRIs
+6. **`user`**: The default evaluation namespace
+
+### The User Namespace
+
+The `user` namespace is special - it contains:
+- All functions from `fluree.db.query.exec.eval` (unqualified)
+- Additional qualified symbols that might be used in queries
+- The `iri` function (both qualified and unqualified versions)
+
+When SCI evaluates expressions without an explicit namespace (like filter expressions in queries), they execute in the `user` namespace by default. This is why most functions are duplicated there.
+
+### Why Multiple Namespaces?
+
+Even though many functions are available in `user`, we still need the other namespaces for:
+- **Qualified symbol resolution**: Expressions like `where/->typed-val` need the `where` namespace
+- **Namespace isolation**: Keeps the code organized as in regular Clojure
+- **Compatibility**: Allows both qualified and unqualified symbol usage
+
+## Key Functions and Patterns
+
+### The `iri` Function
+
+The `iri` function is particularly complex because it needs access to the query context at runtime. In regular Clojure, it's a macro that accesses a context variable. In SCI:
+
+1. A default `iri-fn` is registered that throws an error if called
+2. When evaluating with context, we create a context-aware version
+3. This version is injected into the SCI namespaces via `eval-graalvm-with-context`
+
+### Context Injection Pattern
+
+```clojure
+(defn eval-graalvm-with-context
+  [form ctx]
+  (let [iri-with-context (fn [input] 
+                          ;; context-aware implementation
+                          ...)
+        updated-namespaces (-> current-namespaces
+                              (assoc-in ['user 'iri] iri-with-context)
+                              ;; ... other updates
+                              )]
+    (sci/eval-form (sci/merge-opts @sci-context-singleton
+                                   {:namespaces updated-namespaces
+                                    :bindings {'$-CONTEXT ctx}}))))
+```
+
+### Macro Replacements
+
+Several Clojure macros are replaced with function equivalents for SCI:
+- `if` → `-if-fn`
+- `and` → `-and-fn`
+- `or` → `-or-fn`
+- `as` → `as-fn`
+
+These replacements handle the special evaluation semantics of the original macros.
+
+## Symbol Resolution
+
+### Qualified Symbols
+
+During macro expansion, symbols often get namespace-qualified. The `qualified-symbols` map in eval.cljc maps these qualified symbols to their implementations:
```

Comment:
This isn't quite accurate - `qualified-symbols` maps "non-qualified" symbols to "qualified" symbols, which is how they are identified in the namespace. This map doesn't actually map to implementations, it is strictly symbol to symbol